webhook/validator/isbsvc.go

```diff
@@ -42,17 +43,28 @@ func (v *isbsvcValidator) ValidateCreate(_ context.Context) *admissionv1.Admissi
 }
 
 func (v *isbsvcValidator) ValidateUpdate(_ context.Context) *admissionv1.AdmissionResponse {
+	// check the new ISB Service is valid
 	if err := isbsvccontroller.ValidateInterStepBufferService(v.newISBService); err != nil {
 		return DeniedResponse(err.Error())
 	}
 	switch {
 	case v.oldISBService.Spec.JetStream != nil:
+		// check the type of ISB Service is not changed
 		if v.newISBService.Spec.Redis != nil {
-			return DeniedResponse("Can not change ISB Service type from Jetstream to Redis")
+			return DeniedResponse("can not change ISB Service type from Jetstream to Redis")
+		}
+		// check the persistence of ISB Service is not changed
+		if !equality.Semantic.DeepEqual(v.oldISBService.Spec.JetStream.Persistence, v.newISBService.Spec.JetStream.Persistence) {
+			return DeniedResponse("can not change persistence of Jetstream ISB Service")
 		}
 	case v.oldISBService.Spec.Redis != nil:
+		// check the type of ISB Service is not changed
 		if v.newISBService.Spec.JetStream != nil {
-			return DeniedResponse("Can not change ISB Service type from Redis to Jetstream")
+			return DeniedResponse("can not change ISB Service type from Redis to Jetstream")
+		}
+		// check the persistence of ISB Service is not changed
+		if !equality.Semantic.DeepEqual(v.oldISBService.Spec.Redis.Native.Persistence, v.newISBService.Spec.Redis.Native.Persistence) {
```

Comment:
Need to check if `native` is nil.

Reply:
Thanks! I am adding a nil check before deepequal. Do we need to handle `native` changing from nil to non-nil and vice versa?

Reply:
It's needed.

Reply:
done.